src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -646,6 +646,16 @@ static Linker nativeLinker() {
         /**
          * {@return a linker option used to denote the index of the first variadic argument layout in the
          *          function descriptor associated with a downcall linkage request}
+         * <p>
+         * The {@code index} value must be greater than zero, and less than or equal to the number of argument layouts
+         * of the function descriptor that is used in the same linkage request as this option. When the {@code index} is
+         * equal to the number of argument layouts in the descriptor, it indicates a variadic function to which zero
+         * variadic arguments are passed. It is important to always use this linker option when linking a variadic function,
+         * as this affects the linking process on certain platforms (even if no variadic arguments are passed).
+         *
+         * @implNote The index value is validated when making a linkage request, which is when the function descriptor
+         * against which the index is validated is available.
+         *
          * @param index the index of the first variadic argument layout in the function descriptor associated
```

Comment:
IMHO this description is still confusing because it speaks about an index into the function descriptor argument layouts - which this thing is not.

Reply:
Right, will polish this some more as well.

Reply:
Possible suggestion/thing to try: use a bullet list to spell out all cases for `index`. E.g. we know there's index == 0 (all variadic). Then we know there's index = N (no variadic). Then there's index == m, 0 < m < N - which means layouts 0..m are non-variadic and m..N are variadic (where n..m denotes an interval with n included and m excluded).

Reply:
> Possible suggestion/thing to try: use a bullet list to spell out all cases for `index`. E.g. we know there's index == 0 (all variadic). Then we know there's index = N (no variadic). Then there's index == m, 0 < m < N - which means layouts 0..m are non-variadic and m..N are variadic (where n..m denotes an interval with n included and m excluded).

I think this is a good suggestion. It makes it much easier to understand.

Reply:
Thanks for the review.

I've added a bullet list, and switch same of the language to refer to the 'start of the variadic arguments passed to the function described by the function descriptor'. I think the latter avoids implying the index is an index into the argument layouts. (I've also added a small note to the global variadic function doc to indicate that the index might not necessarily have a corresponding argument layout).

How does the new version look?